assets/src/data/eventespresso/core/model/checkin/resolvers.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * External imports
+ */
+import { getEndpoint } from '@eventespresso/model';
+import { isModelEntityFactoryOfModel } from '@eventespresso/validators';
+import { isEmpty } from 'lodash';
+
+/**
+ * Internal imports
+ */
+import { fetch, dispatch, resolveSelect } from '../../../base-controls';
+import { REDUCER_KEY } from '../../constants';
+import { REDUCER_KEY as SCHEMA_REDUCER_KEY } from '../../../schema/constants';
+
+/**
+ * Resolver for the getLatestCheckin action.
+ *
+ * @param {number} registrationId
+ * @param {number} dateTimeId
+ * @return {BaseEntity|null} Null if there is an error or no checkin record.
+ */
+export function* getLatestCheckin( registrationId, dateTimeId ) {
+	let checkInResponse;
+	const path = `${ getEndpoint( 'checkin' ) }/` +
+		`?where[REG_ID]=${ registrationId }&where[DTT_ID]=${ dateTimeId }` +
+		'&order_by[CHK_timestamp]=DESC&limit=1';
+	try {
+		checkInResponse = yield fetch( {
+			path,
+			method: 'GET',
+		} );
+		if ( isEmpty( checkInResponse ) ) {
+			// there is no checkin record yet!
+			return null;
+		}
+		checkInResponse = checkInResponse.pop();
+	} catch ( error ) {
+		// @todo need to do something different when the user isn't authed and
+		// this is the cause for the error?
+		return null;
+	}
+	const factory = yield resolveSelect(
+		SCHEMA_REDUCER_KEY,
+		'getFactoryForModel',
+		'checkin'
+	);
+	if ( ! isModelEntityFactoryOfModel( factory, 'checkin' ) ) {
+		return null;
```

Comment:
same point as earlier... if the model is invalid we just do... nothing?
maybe there should be some kind of `InvalidModel` exception/error we can throw/return in these cases

Reply:
Just to be clear, this is not a model check.  It is a check for the factory that builds the models.  It'd have to be a fairly significant critical error if the factory isn't available for the given model (i.e. the REST API is not accessible on a site - which in turn would cause problems elsewhere).

If anything, I'm thinking of just removing this check, because if there is no factory available its a significant critical exception and thus the error from the usage of factory in the next line would be sufficient to indicate something wrong to the developer.

I'll think on this though, I might just add a warning after all (as it would mostly be a development issue).

Reply:
added a warning